### DISCUSSION TITLE
Help to type a Model's factory function

### DISCUSSION THREAD

**Discussion by mogarick** [mobxjs/mobx-state-tree#1758] [Q&A]
Hello,

I have a simple factory to create MST models that need to have other objects injected on creation and I do it in the factory. The thing is that it's currently not typed. 

I want to type it but I don't know how to solve it with typescript.

Here is my current simple factory fn

``` javascript
createModelOf({ Type, data }) {
      return Type.create(data,{depInjectionObjA, depInjectionObjB});
    }
```

`Type` is a Model such as `Todo`:

``` javascript
const Todo = types.model({
    name: types.string,
    done: types.boolean
})
```

`data` is the data used to create the model.
`depInjectionObjA` and  `depInjectionObjB` are objects injected from the module where the factory fn resides.

``

So an example call is:

``` javascript
let myTodo = createModelOf({Type:Todo, data:{name:"Do something", done:false}}

let user= createModelOf({Type:User, data{name:"John", lastName:"Doe"}}
```
Could anybody help me to solve this?

Thank you in advance! :D

**Comment by justin-hackin** [comment]
I was looking for a similar solution @mogarick and I noticed this relevant comment in the issues tracker https://github.com/mobxjs/mobx-state-tree/issues/1359#issuecomment-518393477
